pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -138,41 +138,61 @@ async def run(self):
                             self._logger.error(slack_err)
                     raise build_err
 
-                # Create a PR to pin microshift build
+                # Nothing to do for stream builds after this
+                if assembly_type is AssemblyTypes.STREAM:
+                    return
+
+                message = f"microshift for assembly {self.assembly} has been successfully built."
+                await self.slack_say(message)
+
+            # Check if we need create a PR to pin the build
+            pinned_microshift_rpms = util.get_rpm_if_pinned(releases_config, self.assembly, 'microshift')
+            diff = set(nvrs) - set(pinned_microshift_rpms.values())
+            if diff:
+                self._logger.info("Creating PR to pin microshift build: %s", diff)
                 assembly_basis = get_assembly_basis(releases_config, self.assembly)
                 if assembly_basis.brew_event:
                     pr = await self._create_or_update_pull_request(nvrs)
 
-            # Sends a slack message
-            message = f"microshift for assembly {self.assembly} has been successfully built."
-            if pr:
-                message += f"\nA PR to update the assembly definition has been merged: {pr.html_url}"
+                message = (f"PR to pin microshift build to the {self.assembly} assembly has been merged:"
```

Comment:
need to under the if to make sure pr exist

Reply:
if diff exists -> pr should always be created , so if pr doesn't exist then something went wrong - i'm happy to let it fail at that point

Reply:
then why need `if assembly_basis.brew_event:`

Reply:
Good point, I've removed that now - since we are returning early for stream assembly it's not needed. I would like to refactor this in another pr - so we have 2 different funcs for stream and named assemblies so we don't have to do if-else for every step